### ISSUE TITLE
Invalid undefined-variable error for metaclass value in Python 3.5

### ISSUE THREAD

**Issue by mitar** [PyCQA/pylint#1603]
### Steps to reproduce
1. Create two files in a module. `foo.py`:

```python
def fun():
    pass

Metaclass = fun()
```

`test.py`:

```python
from . import foo


class Foo(metaclass=foo.Metaclass):
    pass
```

2. Run pylint on `test.py`.

### Current behavior

Two relevant errors are returned:

```
E:  4, 0: Undefined variable 'foo.Metaclass' (undefined-variable)
W:  1, 0: Unused import foo (unused-import)
```

### Expected behavior

No error is returned about undefined variable `foo.Metaclass` or unused import.

### pylint --version output

```
No config file found, using default configuration
pylint 1.7.2, 
astroid 1.5.3
Python 3.5.2 (default, Oct 11 2016, 05:00:16) 
[GCC 4.2.1 Compatible Apple LLVM 7.0.2 (clang-700.1.81)]
```

**Comment by PCManticore** [comment]
Is this inside a package? It doesn't seem so from your example, which is why this might not work

**Comment by mitar** [comment]
I made a reproduction here:

https://github.com/mitar/pylint-issue-1603

Running `pylint foo/` shows (among others, unrelated) the following:

```
E:  4, 0: Undefined variable 'metaclass.AbstractMetaclass' (undefined-variable)
W:  1, 0: Unused metaclass imported from foo.utils (unused-import)
```
